### PR TITLE
No events on blank space

### PR DESCRIPTION
**Pull request by Dwaynekj**
See [GRID-440](https://appoji.jira.com/browse/GRID-440)

@joneit

### CODE REVIEW

File: src/renderer/index.js [hunk 1]
```diff
@@ -525,15 +525,19 @@ var Renderer = Base.extend('Renderer', {
...
-        // cellEvent.visibleRow = vr;
+        if (vr && vc) {
+            var mousePoint = this.grid.newPoint(x - vc.left, y - vr.top),
+                cellEvent = new this.grid.behavior.CellEvent(vc.columnIndex, vr.index);
 
-        return Object.defineProperty(cellEvent, 'mousePoint', { value: mousePoint });
+            // cellEvent.visibleColumn = vc;
+            // cellEvent.visibleRow = vr;
+
+            return Object.defineProperty(cellEvent, 'mousePoint', {value: mousePoint});
```
Comment (joneit) [inline src/renderer/index.js]: Can we please try whenever possible to stick to the pattern of a single `return` statement at the bottom:
```javascript
function() {
    var result;

    if (condition) {
        result = ...;
    }

    return result;
}
```
Note the falsy return value `undefined` is usually perfectly workable (unless a `null` is specifically required).

File: src/renderer/index.js [hunk 2]
```diff
@@ -525,15 +525,19 @@ var Renderer = Base.extend('Renderer', {
...
+        }
+        return null;
```
Comment (joneit) [inline src/renderer/index.js]: Did you check _all_ usages of `getGridCellFromMousePoint ` to make sure they deal with falsy return?

File: src/lib/events.js [hunk 3]
```diff
@@ -237,17 +237,24 @@ module.exports = {
...
+                    {
+                        value: e,
+                        enumerable: false,
+                        configurable: true,
+                        writable: true
+                    }
+                );
+
+            }
+           if (decoratedEvent) {fn(decoratedEvent);}
```
Comment (Dwaynekj) [inline src/lib/events.js]: duh. Thanks

File: src/renderer/index.js (hunk 1 above)
Comment (Dwaynekj) [inline src/renderer/index.js]: null feels more descriptive about the type signature than undefined to me

Comment (joneit) [inline src/renderer/index.js]: In that case the preferred pattern would be:
```javascript
function() {
    var result = null;

    if (condition) {
        result = ...;
    }

    return result;
}
```
or:
```javascript
function() {
    var result;

    if (condition) {
        result = ...;
    } else {
        result = null;
    }

    return result;
}
```

File: src/renderer/index.js (hunk 2 above)
Comment (joneit) [inline src/renderer/index.js]: Nevermind, I see this call _is the only usage!_

Comment (Dwaynekj) [inline src/renderer/index.js]: ok

File: src/lib/events.js [hunk 4]
```diff
@@ -237,17 +237,24 @@ module.exports = {
...
-                self.getGridCellFromMousePoint(e.detail.mouse),
-                'primitiveEvent',
-                {
-                    value: e,
-                    enumerable: false,
-                    configurable: true,
-                    writable: true
-                }
-            );
+        function handleMouseEvent(e, fn) {
```
Comment (Dwaynekj) [inline src/lib/events.js]: Done

File: src/lib/events.js (hunk 3 above)
Comment (Dwaynekj) [inline src/lib/events.js]: Done

**Pull request merged**